### ISSUE TITLE
Failed to decode param '/%PUBLIC_URL%/favico.ico'

### ISSUE THREAD

**Issue by vitorcamachoo** [nrwl/nx#4762]
## Current Behavior

After upgrade to this latest version, on running application I got:
```shell
event - compiled successfully
[ ready ] on http://localhost:4000
URIError: Failed to decode param '/%PUBLIC_URL%/favicon.ico'
    at decodeURIComponent (<anonymous>)
    at decode_param (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/layer.js:172:12)
    at Layer.match (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/layer.js:123:27)
    at matchLayer (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:574:18)
    at next (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:220:15)
    at SendStream.error (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/serve-static/index.js:121:7)
    at SendStream.emit (node:events:376:20)
    at SendStream.error (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/send/index.js:270:17)
    at SendStream.pipe (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/send/index.js:520:10)
    at serveStatic (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/serve-static/index.js:125:12)
    at Layer.handle [as handle_request] (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:317:13)
    at /Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:275:10)
    at expressInit (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/middleware/init.js:40:5)
    at Layer.handle [as handle_request] (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:317:13)
    at /Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/router/index.js:275:10)
    at query (/Users/vitorcamacho/workspace/dashboard-HQ/node_modules/express/lib/middleware/query.js:45:5)
```
## Steps to Reproduce

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Environment
```shell
MacOS M1 

npm: 7

@nrwl/cli                 11.1.5   →    11.2.12 
  @nrwl/eslint-plugin-nx   ^11.1.5   →   ^11.2.12 
  @nrwl/jest                11.1.5   →    11.2.12 
  @nrwl/next                11.1.5   →    11.2.12 
  @nrwl/react               11.1.5   →    11.2.12 
  @nrwl/workspace           11.1.5   →    11.2.12 
```

**Comment by mandarini** [comment]
Hi there @vitorcamachoo ! Thanks for filing an issue! I tried to reproduce your issue, but I could not.

Let me tell you the steps I followed. 
Here is also a repo of these steps, all in individual commits: https://github.com/mandarini/test-4762
I can run my app both before and after migration.

Steps:
```
npx create-nx-workspace@11.1.5 test-4762
# I chose to generate a nextjs workspace with a nextjs app named test-app
cd test-4762
# I also generated a react app - to test both
nx generate @nrwl/react:application --name=other-test-app
nx run test-app:serve
# SUCCESS
nx run other-test-app:serve
#SUCCESS
nx migrate @nrwl/workspace@11.2.12
yarn
nx migrate --run-migrations
nx run test-app:serve
# SUCCESS
nx run other-test-app:serve
#SUCCESS
```

**Comment by vitorcamachoo** [comment]
Hi @mandarini 

Thanks for your help!

I checked that you used yarn, did you tried with npm 7 ?

**Comment by vitorcamachoo** [comment]
Seems like removing old dist folder solves the problem. Can be related with the cache folder.

**Comment by mandarini** [comment]
Hi @vitorcamachoo ! I closed this issue, since you seemed to solve it. I may be related to the cache folder, or potentially to leftover artefacts in the dist folder (since deleting it solves the issue). Let me know if it persists and we can reopen! Thank you! :)

**Comment by rahul-sharma-uipath** [comment]
+1 on this issue 

@mandarini we have a case where some of our assets (favicon, manifest, etc) need to be hosted on a CDN and referenced via the index.html and currently there's no support for that in this library.

I know the library does support the use of a `deployUrl` but I wonder if we can extend the concept to some environment variable similar to PUBLIC_URL e.g NRWL_DEPLOY_URL which is replaced at build time with the webpack configuration. That way, in the index.html, we can prefix any files that are deployed on a CDN with this env var and webpack will replace it at build time

**Comment by redpredator8** [comment]
Fix: if you have github, revert to the last change you did / unistall recent libraries

I updated my mac and this started happening, was killing my brain tbh. Im pretty sure it was bc i installed firebase tools and that messed up something because i updated my mac and maybe smth didnt save by mac time-machine so it was throwing a weird hidden syntax error that was preventing the app to be displayed. I tried fixing all the errors like the guys up top are saying fixing the links etc but brand new shit was popping up and I was like nahh. no way i have to recode Node to fix this shit.

Options: Revert to previous version / uninstall recent installed libraries / Create a new react app and just copy your src folder to the new app

leave a thubs up: )
